airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlSource.java

```diff
@@ -313,6 +331,46 @@ public Set<String> getExcludedInternalNameSpaces() {
         "sys");
   }
 
+  @Override
+  protected boolean verifyCursorColumnValues(final JdbcDatabase database, final String schema, final String tableName, final String columnName)
+          throws SQLException {
+    boolean nullValExist = false;
+    final String resultColName = "nullValue";
+    final String descQuery = schema == null || schema.isBlank()
+            ? String.format(DESCRIBE_TABLE_WITHOUT_SCHEMA_QUERY, tableName)
+            : String.format(DESCRIBE_TABLE_WITH_SCHEMA_QUERY, schema, tableName);
+    final Optional<JsonNode> field = database.bufferedResultSetQuery(conn -> conn.createStatement()
+                            .executeQuery(descQuery),
+                    resultSet -> JdbcUtils.getDefaultSourceOperations().rowToJson(resultSet))
+            .stream().filter(x -> x.get("Field").asText().equalsIgnoreCase(columnName))
+            .findFirst();
+    if(field.isPresent()){
+      final JsonNode jsonNode = field.get();
+      final JsonNode isNullable = jsonNode.get("Null");
+      if (isNullable!=null){
+        if (isNullable.asText().equalsIgnoreCase("YES")){
+          final String query = schema == null || schema.isBlank()
+                  ? String.format(NULL_CURSOR_VALUE_WITHOUT_SCHEMA_QUERY,
+                  tableName, columnName, resultColName)
+                  : String.format(NULL_CURSOR_VALUE_WITH_SCHEMA_QUERY,
+                  schema, tableName, columnName, resultColName) ;
+
+          LOGGER.debug("null value query: {}", query);
+          final List<JsonNode> jsonNodes = database.bufferedResultSetQuery(conn -> conn.createStatement().executeQuery(query),
+                  resultSet -> JdbcUtils.getDefaultSourceOperations().rowToJson(resultSet));
+          Preconditions.checkState(jsonNodes.size() == 1);
+          nullValExist = convertToBoolean(jsonNodes.get(0).get(resultColName).toString());
+          LOGGER.debug("null value exist: {}", nullValExist);
```

Comment:
nit : Only log if `nullValExist` = true. and update message to "null cursor value exists for MySQL source"